src/main/java/org/openrewrite/java/migrate/lang/UseVarKeyword.java

```diff
@@ -0,0 +1,162 @@
+import org.openrewrite.Cursor;
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.internal.lang.Nullable;
+import org.openrewrite.java.JavaParser;
+import org.openrewrite.java.JavaTemplate;
+import org.openrewrite.java.JavaVisitor;
+import org.openrewrite.java.search.HasJavaVersion;
+import org.openrewrite.java.tree.Expression;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.JavaType;
+import org.openrewrite.java.tree.TypeTree;
+
+import java.time.Duration;
+
+import static java.lang.String.format;
+import static java.util.Objects.*;
+
+@Value
+@EqualsAndHashCode(callSuper = false)
+public class UseVarKeyword extends Recipe {
+    public String getDisplayName() {
+        return "Use local variable type-inference (var) where possible";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Local variable type-inference reduce the noise produces by repeating the type definitions in Java 10 or higher.";
+    }
+
+    @Override
+    public @Nullable Duration getEstimatedEffortPerOccurrence() {
+        return Duration.ofMinutes(1);
+    }
+
+    @Override
+    protected TreeVisitor<?, ExecutionContext> getSingleSourceApplicableTest() {
+        return new HasJavaVersion("10", true).getVisitor();
+    }
+
+    @Override
+    protected TreeVisitor<?, ExecutionContext> getVisitor() {
+        // J.VariableDeclarations
+        return new JavaVisitor<ExecutionContext>() {
+            private final JavaType.Primitive SHORT_TYPE = JavaType.Primitive.Short;
+            private final JavaType.Primitive BYTE_TYPE = JavaType.Primitive.Byte;
+            private final JavaTemplate template = JavaTemplate.builder(this::getCursor, "var #{} = #{any()}")
+                    .javaParser(JavaParser.fromJavaVersion()).build();
+
+            @Override
+            public J.VariableDeclarations visitVariableDeclarations(J.VariableDeclarations multiVariable, ExecutionContext executionContext) {
+                J.VariableDeclarations vd = (J.VariableDeclarations) super.visitVariableDeclarations(multiVariable, executionContext);
+
+                boolean isOutsideMethode = !determineIfIsInsideMethode(this.getCursor());
+                boolean isMethodeParameter = determineIfMethodeParameter(vd, this.getCursor());
+                if (isOutsideMethode || isMethodeParameter) return vd;
+
+                TypeTree typeExpression = vd.getTypeExpression();
+                boolean isByteVariable = typeExpression instanceof J.Primitive && BYTE_TYPE.equals(typeExpression.getType());
+                boolean isShortVariable = typeExpression instanceof J.Primitive && SHORT_TYPE.equals(typeExpression.getType());
+                if (isByteVariable || isShortVariable) return vd;
+
+                boolean definesNoVariable = vd.getVariables().isEmpty();
+                boolean isCompoundDefinition = vd.getVariables().size() < 1;
```

Comment:
```suggestion
                boolean isCompoundDefinition = vd.getVariables().size() > 1;
```

Also, I suggest to add a corresponding test.

Reply:
Yes totally! And as well, for your suggestion about `for(var s : List.of("Hello", "world", "!")) { /* ...*/ }` 
I've checked the JLS, and it seems like I misused the term "compound" see [JLS SE 8: 16.26.2](https://docs.oracle.com/javase/specs/jls/se8/html/jls-15.html#jls-15.26.2) combine it to `boolean definedSingleVariable = vd.getVariables().size() == 1;`. I think it is better in terms of semantic and naming ;)